### ISSUE TITLE
map update in ensures 

### ISSUE THREAD

**Issue by jiplucap** [Microsoft/dafny#145]
The notation m[t := u]  is not allowed in ensures clauses in Dafny 2.0.0?
It says that t is "undeclared identifier" while it is a parameter of the method where the ensures clause is.

**Comment by wilcoxjay** [comment]
It appears to work for me. Can you provide a minimum working example?

**Comment by jaylorch** [comment]
If you find an example, please reopen this issue.

**Comment by alexlesaka** [comment]
I'm working with @jiplucap and we found the following example: https://rise4fun.com/Dafny/Rvv6

I included 2 modules: in "aux" I wrote functions, predicates and datatypes that are used in the lemmas where we found problems. In the other one ("problem") there are two lemmas that cause the error that @jiplucap said.

We didn't have problems with previews version of Dafny but with Dafny 2.0.0 there are.

Apparently the problem is caused when _setOfValModelsNoFreeForall_Lemma_  calls inside _forall a,f_ to _removeVar_Lemma_. If you deleted the f from the forall that is inside, the problem desappeares (**forall a,f** --> **forall f** ). 

We don't know if it is a bug or it is something that we don't understand and it works as you expected. 

Thank you in advance.
